dace/transformation/transformation.py

```diff
@@ -164,12 +230,19 @@ def __lt__(self, other):
             if self_end:
                 return True
 
-    def apply_pattern(self, sdfg):
-        """ Applies this transformation on the given SDFG. """
-        sdfg.append_transformation(self)
-        self.apply(sdfg)
-        if not self.annotates_memlets():
-            propagation.propagate_memlets_sdfg(sdfg)
+    @classmethod
+    def _get_pattern_nodes(cls) -> Dict[str, 'PatternNode']:
+        """ 
+        Returns a dictionary of pattern-matching node in this transformation
+        subclass. Used internally for pattern-matching.
+        :return: A dictionary mapping between pattern-node name and its type.
+        """
+        return {
+            k: getattr(cls, k)
+            for k in dir(cls)
+            if isinstance(getattr(cls, k), PatternNode) or (k.startswith(
```

Comment:
There was a ruse in google style guide that suggests avoiding multiline conditions in comprehensions. Can we extract the condition into a method?

Reply:
This would make the code unnecessarily longer IMO.

Reply:
Longer is better if it is more readable, feel free to leave as is if not agree.